src/main/io/serial_cli.c

```diff
@@ -2800,6 +2833,34 @@ static void printConfig(char *cmdline, bool doDiff)
     }
 
     createDefaultConfig(&defaultConfig);
+    // make copies of configs to do differencing
```

Comment:
PGs may place all configs into separate section (in ram). Then it will be trivial to  backup whole section (and using offset, call all diffing functons) ...

Another possibility is to iterate all PGs (or only selected one, may be enumerated by PGID), appending copy to buffer (buffer size is a bit problematic) and storing pointers for diff functions.